File: app/src/main/java/com/vmenon/mpo/api/Show.java
package com.vmenon.mpo.api;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Index;
import android.arch.persistence.room.PrimaryKey;
import android.support.annotation.NonNull;

import java.util.ArrayList;
import java.util.List;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import org.parceler.Parcel;

@Parcel
@Entity(indices = {@Index(value = "name", unique = true)})
public class Show {

    @NonNull
    @SerializedName("name")
    @Expose
    public String name;

    @SerializedName("artworkUrl")
    @Expose
    public String artworkUrl;

    @SerializedName("genres")
    @Expose
    public List<String> genres = new ArrayList<>();

    @SerializedName("author")
    @Expose
    public String author;

    @SerializedName("feedUrl")
    @Expose
    public String feedUrl;

    @PrimaryKey(autoGenerate = true)
    public long id;

    public long lastUpdate = -1L;
    public long lastEpisodePublished = -1L;
}

File: app/src/main/java/com/vmenon/mpo/api/Show.kt
package com.vmenon.mpo.api

import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

import java.util.ArrayList
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

import org.parceler.Parcel

@Parcel
@Entity(indices = [Index(value = ["name"], unique = true)])
data class Show(
    @SerializedName("name")
    @Expose
    val name: String = "",

    @SerializedName("artworkUrl")
    @Expose
    val artworkUrl: String? = null,

    @SerializedName("genres")
    @Expose
    val genres: List<String> = ArrayList(),

    @SerializedName("author")
    @Expose
    val author: String? = null,

    @SerializedName("feedUrl")
    @Expose
    val feedUrl: String? = null,

    @PrimaryKey(autoGenerate = true)
    val id: Long = 0L,

    var lastUpdate: Long = -1L,
    var lastEpisodePublished: Long = -1L
)
